### PR TITLE
tresoldi_multitiers submission

### PR DESCRIPTION
**Pull request by tresoldi**
This PR includes a system with my submission, following the directions given in the documentation. All changes are contained in the `systems/tresoldi_multitiers` directory.

It includes instructions for reproducing the analysis, the results of the prediction, and reports on accuracy. The actual models for prediction are not included, as they are rather big and, in many cases, would be over the maximum file size allowed by GitHub. They are available as `.tar.gz` files from links provided in the `README.md` file.

The system follows a multitiered approach, developing on what previously discussed by List and Chacon (2015) and Tresoldi et al. (2018). It is more of an exploratory approach on the suitability of the idea, which would be explained in more detail in the paper submission.

### CODE REVIEW

**Comment by LinguList** [comment]
@tresoldi, thanks, may I please ask you to separate the training and surprise data, as I did in my example? You have now all data in a single folder, but I want the results to be separated.

**Comment by LinguList** [comment]
In fact, I don't know if I can see the whole directory structure, due to the huge number of files, etc., and GH keeps not showing all. So what I'd like to see there is specifically the structure:

```
tresoldi-multitiers
    surprise
        bantubvd
            results-0.10.tsv
        ...
    training
        allenbai
            results-0.10.tsv
```

Is this clear, maybe you even have it there? In your submission, you list many of the files that we provide (training, etc.), which may not be needed, if you access them with the library we deliver, right?

And one more point: do we need a `.gitignore` file in the systems files?

**Comment by tresoldi** [comment]
Thanks @LinguList . On the topics:

- Separating training and surprise: sure, that is easy. Unfortunately I will only be able to do it on the deadline of Monday (I'm on a trip now, no proper access to a computer until then)
- Large number of files: you are right, they were showing when I prepared the PR (or at least I remember them) but the list is now missing, even for me. You can browse the files on my fork ( https://github.com/tresoldi/ST2022/tree/main/systems/tresoldi_multitiers ), but the solution will likely be for me to drop all the files in `dfs/` and `mtdata`, perhaps with a zipped copy somewhere. I added them more as a convenience, but they are quick to rebuild (takes about five minutes for all datasets) and they are fully reproducible everywhere (unlike the stochastic models). But it will have to wait for Monday as well.
- gitignore: It is not necessary, I just added some entries to get a clean list without the serialized models, cluster scripts, etc. I can delete if you prefer.

**Comment by tresoldi** [comment]
My comment was not clear: there are no copies of the files you provide, but similarly named, intermediate files with the extended information. These are the ones built with the first script and which I will drop -- they are there so that different tools (like ML libraries for R and Julia) can be used (which was the plan but I did not finish).

**Comment by LinguList** [comment]
Okay, I see, thanks for the clarification. So I think you should decide which level of granularity you want to have there. I for my part do not really need intermediate files, but the output with training and surprise, as I said, and I would later take time to try and test the systems, but not upon submission now (as I do not have time myself, unfortunately).

So I'll wait for you to indicate when you are finalized and merge then.

**Comment by tresoldi** [comment]
Ok, perfect. I will do it Monday morning, perhaps even Sunday evening.

BTW, no problem on dividing training and surprise in different directories, but the `st2022` seems to accept all in the same directory without problems, here are the outputs: https://github.com/tresoldi/ST2022/tree/main/systems/tresoldi_multitiers/reports

**Comment by LinguList** [comment]
Yes, I know, but I still want them in separate repositories, so we have 
same paths ;)

**Comment by tresoldi** [comment]
@LinguList PR is now updated. I have:

- renamed the submission, following the name pattern in other systems
- deleted all intermediate files, except for all results in all different models (they are all in `output/`, and I can remove them as well)
- added a script to structure the results in the expected way, in `training/` and `surprise/`
- updated my README root file

**Comment by tresoldi** [comment]
I had forgotten .gitignore, it is now deleted as well.

**Review by LinguList — CHANGES_REQUESTED** [review]
@tresoldi, where do I find the folder `surprise` and where the folder with the training data results? Can I please ask you to check again with my example to mirror the structure there? And if you have several models, please separate them, as we discussed in a previous PR (please check the one by team Mockingbird).

**Comment by tresoldi** [comment]
@LinguList The `surprise` and `training` directories could be created with the new script, I hadn't pushed them because it would mean two copies of the files. But I see that GitHub can handle the number of files now, so I just added them.

There are different models, but there is only one submission, named `best` in the `output` directory. The idea is to experiment with different models and in each case pick the one expected to perform the best (based on the stratified sampling during hyperparameter search on training). In the envisioned scenario they would be part of an ensemble weighted by such an evaluation (so that it is really a single model), but there was no computational power to finish this.

**Review by LinguList — APPROVED** [review]
Thanks.

**Comment by tresoldi** [comment]
Thinking about the issue, I understand this might seem "unfair" in the evaluation. If it is ok with you, I will prepare a second submission with the data from the only model I had time to fully run, using random forests (naming it `extalign_rf`).

The overall performance is worse as one would expect (LGB and MLP were in general much better), but it should give a better idea of the performance on a quick-to-train machine learning method, and how much one could gain from more computation spent on the training.
